examples/front.kk

```diff
@@ -0,0 +1,105 @@
+/*----------------------------------------------------------------------------
+   Copyright 2024, Koka-Community Authors
+
+   Licensed under the MIT License ("The License"). You may not
+   use this file except in compliance with the License. A copy of the License
+   can be found in the LICENSE file at the root of this distribution.
+----------------------------------------------------------------------------*/
+
+import html/html
+
+import html/layout
+
+
+// todo: this deserves a PR to go into std/core?
+fun do(action : () -> e a) : e a
+  action()
+
+
+value struct cnode
+  swarm : string
+  ip : string
+  ncores : int
+  avail-ram : int
+  total-ram : int
+
+effect val cnodes-info : list<cnode>
+
+fun front() : <cnodes-info> component
+
+  /* todo: a PR to introduce the `super` keyword, to be used in record-updates?
+          val page = basic-template(page-head = {
+            page-head(super)()
+            link("stylesheet", "text/css", "/front.css")
+          })
+  */
+  val tmpl = basic-template(page-head = {
+    page-head(basic-template)()
+    link("stylesheet", "text/css", "/front.css")
+  })
+
+  with create-component <- layout-html(tmpl)
+  // customization to the page template follows
+
+  with val page-title = "The Front Page"
+
+  with fun body-content() // admit no content from the template, fresh new here
+    h3
```

Comment:
The root of the issue stems from here.
The problem is that `h3` requires the `html-tree` effect but `create-component : (() -> <html-tmpl|e> component) -> e component` has `e` as the underlying effect. This causes a unification with `html-tree` which is not handled above this handler. (Remember that handler functions are run after yielding to the context of their handler).

Bidirectional effects (which Koka doesn't have, but we could add a feature request for), as far as I understand them allow you to state that the effect function should be called in the context of the call to the effect, and not first yield up to the handler. In this way some effects from the call site can be used. (And `html-tree` would be handled in this case). 

My first attempt at solving the typing problem here, was to change the handlers to use `val body-context = fn() ...` which was sort of a workaround to try to do bidirectional effects in Koka. The problem with this approach was with the polymorphic effect. We can continue trying this approach if that is what you wanted.

The final solution in #3 (sorry for not pushing the code up earlier), changes the design a bit so that the `body-content` and other functions return a component rather than using the effect directly. I've added another version of the component API which handles the `html-tree` without overriding the effect allowing you to build up a subpiece of the tree. Which was probably a good addition anyways. There are a few things about that API that might not have been as clean as your current attempt, but at least it works and type checks. Anyways, I would love to get your feedback there if those API changes seem acceptable. (And it looks like you added to the example an external effect here which I'll have to migrate over).